lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -119,8 +119,8 @@ def add_address(post, card, options)
         address_object[:state] = address[:state] if address[:state]
         address_object[:city] = address[:city] if address[:city]
         address_object[:zip_code] = address[:zip] if address[:zip]
-        address_object[:street] = address[:street] || parse_street(address) if parse_street(address)
-        address_object[:number] = address[:number] || parse_house_number(address) if parse_house_number(address)
+        address_object[:street] = address[:street] || parse_street(address) if ( address[:address1] && parse_street(address) )
+        address_object[:number] = address[:number] || parse_house_number(address) if ( address[:address1] && parse_house_number(address) )
```

Comment:
I feel like these parse methods should include a guard clause instead of running it as part of the conditional here.

Reply:
Thanks, updated 👍